### PR TITLE
Change folder structures, set up entities/graphql structures

### PR DESCRIPTION
**Pull request by minamorl**
## About
This PR is intended to show the basic structures to other devs.
Both QueryRoot/Mutation is correctly working under this PR.

### CODE REVIEW

File: src/setup.rs [hunk 1]
```diff
@@ -9,38 +9,38 @@ use sea_orm::*;
 
 pub(super) async fn set_up_db() -> Result<DatabaseConnection, DbErr> {
     dotenv().ok();
-    let DATABASE_URL: std::string::String = env::var("DATABASE_URL").unwrap();
-    let DATABASE_NAME: std::string::String = env::var("DATABASE_NAME").unwrap();
-    let db = Database::connect(DATABASE_URL.to_string()).await?;
+    let database_url: std::string::String = env::var("DATABASE_URL").unwrap();
```
Comment (chansuke) [inline src/setup.rs:12]: unwrap() just panics when you failed to get the environment variable in here, so should choose another way(ex: unwrap_or_else() etc).

**Review by chansuke — CHANGES_REQUESTED** [review]
I've added some comments.

File: src/setup.rs (hunk 1 above)
Comment (minamorl) [reply src/setup.rs:12]: ok tyty

File: src/main.rs [hunk 2]
```diff
@@ -1,60 +1,71 @@
...
         let db = ctx.data::<DatabaseConnection>().unwrap();
         Persona::find_by_id(id).one(db).await
     }
-    async fn personas(&self, ctx: &Context<'_>, id: i32) -> Result<Vec<persona::Model>, DbErr> {
+    async fn personas(&self, ctx: &Context<'_>, _id: i32) -> Result<Vec<persona::Model>, DbErr> {
         let db = ctx.data::<DatabaseConnection>().unwrap();
         Persona::find().all(db).await
     }
+    async fn activities(&self, ctx: &Context<'_>) -> Result<Vec<post::Model>, DbErr> {
+        let db = ctx.data::<DatabaseConnection>().unwrap();
```
[the remark this answers is not in the document]
Comment (minamorl) [reply src/main.rs:48]: using Context lgtm. ill make changes

[the remark this answers is not in the document]
Comment (minamorl) [reply src/main.rs:48]: ref https://async-graphql.github.io/async-graphql/en/context.html

**Comment by minamorl** [comment]
@chansuke Can you take a look at these changes?

**Comment by chansuke** [comment]
@minamorl Ok

File: entities/src/board.rs [hunk 3]
```diff
@@ -0,0 +1,41 @@
+//! `SeaORM` Entity. Generated by sea-orm-codegen 0.10.6
+
+use sea_orm::{entity::prelude::*, Set};
+use ulid::Ulid;
+
+#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq)]
+#[sea_orm(table_name = "board")]
+pub struct Model {
```
Comment (chansuke) [inline entities/src/board.rs:8]: The naming should be `Board` or `BoardModel`?

Comment (chansuke) [reply entities/src/board.rs:8]: ah, > Generated by sea-orm-codegen

File: graphql/src/models.rs [hunk 4]
```diff
@@ -0,0 +1,208 @@
+use async_graphql::*;
```
Comment (chansuke) [inline graphql/src/models.rs:1]: You should separate this file IMHO
- graphql/src/user.rs
etc

File: graphql/src/mutations.rs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+#[Object]
+impl Mutation {
+    async fn create_board(
+        &self,
+        context: &Context<'_>,
+        title: String,
+        content: String,
+    ) -> anyhow::Result<Option<Board>> {
+        let db = context.data::<DatabaseConnection>().unwrap();
```
Comment (chansuke) [inline graphql/src/mutations.rs:18]: remove `unwrap()`  -> let db = context.data::<DatabaseConnection>();

and use map_err() in L: 26

File: migration/README.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
+# Running Migrator CLI
```
Comment (chansuke) [inline migration/README.md:1]: we can consider to use cargo make or Makefile(should be in another issue)

File: graphql/src/queries.rs [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+use jsonwebtoken::{DecodingKey, Validation};
+use sea_orm::{ColumnTrait, DatabaseConnection, EntityTrait, QueryFilter};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct QueryRoot;
+
+#[Object]
+impl QueryRoot {
+    async fn user(&self, context: &Context<'_>, id: String) -> anyhow::Result<Option<User>> {
+        let db = context.data::<DatabaseConnection>().unwrap();
```
Comment (chansuke) [inline graphql/src/queries.rs:15]: same in here. avoid unwrap() in db connection

File: graphql/src/queries.rs [hunk 8]
```diff
@@ -0,0 +1,44 @@
+use crate::models::{Post, User};
+use crate::{Claims, Token};
+use anyhow;
```
Comment (chansuke) [inline graphql/src/queries.rs:3]: anyhow is simple and idiomatic, but better to avoid to use in this layer IMHO.

File: src/lib.rs [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::env;
+    #[tokio::test]
+    async fn it_works() {
+        dotenv::dotenv();
+        let database_url: std::string::String = env::var("DATABASE_URL").unwrap();
```
Comment (chansuke) [inline src/lib.rs:55]: please pass lint and test...

**Pull request merged**